Code/Framework/AzCore/AzCore/Console/ConsoleDataWrapper.inl

```diff
@@ -37,7 +37,13 @@ namespace AZ
         // Client code can supply a type for m_value that overrides the operator= function and trigger side effects
         // in the operator= function body. Doing the assignment outside the value change check avoids those side
         // effects not being triggered because AzCore believes the value wouldn't change.
+
+#if !defined(_RELEASE) && defined(CARBONATED)
```

Comment:
I suggest Release to function same as non-release, the less differences we have the better. Please analyze the direct cvar assignment frequency.

Reply:
Let wait for Lloyd's feedback on RELEASE condition.

Reply:
I agree w/ @aefimov3 here.  We want consistent behaviour in all configurations.